include/emds/emds.h

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ *  @defgroup emds Emergency Data Storage
+ *  @ingroup emds
+ *  @{
+ *  @brief Emergency Data Storage API
+ */
+
+#ifndef EMDS_H__
+#define EMDS_H__
+
+#include <stddef.h>
+#include <sys/types.h>
+#include <sys/util.h>
+#include <sys/slist.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @struct emds_entry
+ *
+ * Information about entries used to store data in the emergency data storage.
+ */
+struct emds_entry {
+	/** Unique id for each static and dynamic entry. */
+	uint16_t id;
+	/** Pointer to data that shall be stored. */
+	uint8_t *data;
+	/** Length of data that shall be stored. */
+	size_t len;
+}
+
+/**
+ * @struct emds_dynamic_entry
+ *
+ * Entries with a node element, used for dynamic entries. These are registered
+ * using a call to emds_entry_add().
+ */
+struct emds_dynamic_entry {
+	struct emds_entry entry;
+	sys_snode_t node;
+}
+
+/**
+ * @brief Define a static entry for emergency data storage items.
+ *
+ * @param name The entry name.
+ * @param _id Unique id for entry. This value and not overlap with any other
+ *            value.
```

Comment:
```suggestion
 * @param _id Unique ID for the entry. This value and not an overlap with any other
 *            value.
```
Not sure if I am getting the purpose of the second part. Can it be written: This value does not overlap with any other value?